### ISSUE TITLE
Provide more information about the build process with --version

### ISSUE THREAD

**Issue by buti1021** [minetest/minetest#14035]
### Problem

To understand a binarys behaviour it is important to understand how it was built. Minetest includes many option, howevery only a view are shown when running minetest or minetest --version.

### Solutions

Include as much information as possible in --version or atleast specify why some information should be included and other not.

### Alternatives

Keep it as is

### Additional context

Example output of --version:

> Minetest 5.6.1 (Linux)
> Using LuaJIT 2.1.0-beta3
> BUILD_TYPE=Release
> RUN_IN_PLACE=1
> USE_CURL=1
> STATIC_SHAREDIR="."
> 

Why is the information about curl included but other information e.g., about databases not?

Relevant code:

https://github.com/minetest/minetest/blob/6783734612832bd6a25adc3ddcd32f6a12c89249/src/main.cpp#L430-L443

and

https://github.com/minetest/minetest/blob/771da80bbb85c9fa4f0dcf0f3d04f46a0a20239f/src/version.cpp#L36-L48

**Comment by rollerozxa** [comment]
Yes, it would be quite useful for `--version` to list all compile options Minetest was built with. For e.g. server owners it is useful to know if a given database backend they want to use has actually been compiled with it enabled (since CMake options can be misleading since they can be enabled, yet will automatically disable if the library and its headers aren't found).

**Comment by wsor4035** [comment]
wasnt really any good label choices for this issue, so picked the closest one in my opinion, other triager/core devs feel free to change, unsubscribing
